tests/testthat/test-get_flusight_bin_endpoints.R

```diff
@@ -216,3 +216,223 @@ test_that("get_flusight_bin_endpoints works, 2023/24", {
+  # - construct data that should fall into known categories
+  #   (with every horizon/category combination, and both criteria for stable)
+  # - compute bins and apply them to the data
+  # - check that we got the right answers
+  location_meta <- readr::read_csv(
+    file = testthat::test_path("fixtures", "location_meta_24.csv")
+  ) |>
+    dplyr::mutate(
+      pop100k = .data[["population"]] / 100000
+    )
+
+  # locations for testing "stable", "increase" and "decrease" thresholds are:
+  # 56 = Wyoming, pop100k = 5.78
+  # 02 = Alaska, pop100k = 7.11 and 11 = District of Columbia, pop100k = 6.69
+  # these states trigger the "minimum count change at least 10" rule
+  locs <- c("US", "56", "02", "11", "05", "06")
+
+  # create data
+  # our reference date will be 2024-11-23.
+  # changes are relative to 2024-11-16.
+  # 2024-11-09 is throw-away, to make sure we grab the right "relative to" date
+  target_data <- tidyr::expand_grid(
+    location = locs,
+    date = as.Date("2024-11-09") + seq(from = 0, by = 7, length.out = 6),
+    value = NA
+  )
+
+  expected_categories <- NULL
+
+  # all expected category levels are "stable": rate change less than
+  # 0.3, 0.5, 0.7, or 1 * population rate
+  loc_pop100k <- location_meta$pop100k[location_meta$location == locs[1]]
+  target_data$value[target_data$location == locs[1]] <- c(
+    0,
+    10000,
+    10000 + floor(0.29 * loc_pop100k),
+    10000 - floor(0.49 * loc_pop100k),
+    10000 + floor(0.69 * loc_pop100k),
+    10000 - floor(0.99 * loc_pop100k)
+  )
+  expected_categories <- dplyr::bind_rows(
+    expected_categories,
+    target_data |>
+      dplyr::filter(
+        .data[["location"]] == locs[1],
+        .data[["date"]] >= "2024-11-23"
+      ) |>
+      dplyr::mutate(
+        horizon = as.integer((.data[["date"]] - as.Date("2024-11-23")) / 7),
+        output_type_id = "stable"
+      )
+  )
+
+  # all expected category levels are "stable": count change less than 10
+  target_data$value[target_data$location == locs[2]] <- c(0, 300, 304, 297, 309, 291)
+  expected_categories <- dplyr::bind_rows(
+    expected_categories,
+    target_data |>
+      dplyr::filter(
+        .data[["location"]] == locs[2],
+        .data[["date"]] >= "2024-11-23"
+      ) |>
+      dplyr::mutate(
+        horizon = as.integer((.data[["date"]] - as.Date("2024-11-23")) / 7),
+        output_type_id = "stable"
+      )
+  )
+
+  # all expected category levels are "increase": count change >= 10,
+  # horizon 0: 0.3 <= rate change < 1.7
+  # horizon 1: 0.5 <= rate change < 3
+  # horizon 2: 0.7 <= rate change < 4
+  # horizon 3: 1.0 <= rate change < 5
+  # note, loc_pop100k for this location is 7.11 < 10
+  loc_pop100k <- location_meta$pop100k[location_meta$location == locs[3]]
+  target_data$value[target_data$location == locs[3]] <- c(
+    0,
+    10000,
+    10000 + 10,
+    10000 + floor(2.99 * loc_pop100k),
+    10000 + floor(3.99 * loc_pop100k),
+    10000 + ceiling(1.01 * loc_pop100k)
```

Comment:
Likewise, I just modified the numbers here, but they didn't seem to be right in the original code (lines 113 to 115)